contrib/epee/include/readline_buffer.h

```diff
@@ -19,12 +21,23 @@ namespace rdln
     }
     void get_line(std::string& line) const;
     void set_prompt(const std::string& prompt);
+    static void add_completion(const std::string& command)
+    {
+      if(std::find(completion_commands.begin(), completion_commands.end(), command) != completion_commands.end())
+	return;
+      completion_commands.push_back(command);
```

Comment:
Ah but here is your misunderstanding. This vector is not a history of commands entered. Readline has its own history. This vector is only used to store the command names of the domain / wallet and is only added to when they construct (or more accurately bind functions to commands at startup). Therefore this vector is always small as it only ever contains the complete list of available commands a user could type - just the commands, not even the arguments.